zulip_bots/zulip_bots/bots/joinme/joinme.py

```diff
@@ -0,0 +1,55 @@
+# See readme.md for instructions on running this code.
+
+from typing import Any
+import requests
+import json
+import socket
+
+class JoinmeHandler(object):
+    def usage(self) -> str:
+        return '''
+        A bot that will post a link to joinme call.
+        '''
+
+    def handle_message(self, message: Any, bot_handler: Any) -> None:
+        if 'ZISSHUDBUNIq' not in message['content']:
+            payload = {'client_id':'fvg6qhpans4z2jryq45crb69', 'scope':'user_info scheduler start_meeting',
+            'redirect_uri':'https://developer.join.me/io-docs/oauth2callback', 'state':'ZISSHUDBUNIq', 'response_type':'code'}
+            r = requests.get('https://secure.join.me/api/public/v1/auth/oauth2', params=payload)
+            content = "Please click the link below to log in and authorise Joinme:\n {} \nAnd please copy and send the url shown in browser after clicking accept. Don't forget to @ me!".format(r.url) # type: str
+            bot_handler.send_reply(message, content)
+        if 'ZISSHUDBUNIq' in message['content']:
+            code1 = message['content']
+            code1 = code1.split('=')
+            code2 = code1[1].split('&')

```

Comment:
Here, I would recommend making `code1` and `code2` more descriptive. Name the variables based on what they are supposed to contain. :)